### PR TITLE
Fix gradle deprecation warning

### LINKED ISSUE #457: Format task show deprecation messages in Gradle 7.0
Calling `./gradlew ktlintFormat` with Gradle 7.0-milestone-3 displays:
```
> Task :runKtlintFormatOverMainSourceSet
Execution optimizations have been disabled for task ':runKtlintFormatOverMainSourceSet' to ensure correctness due to the following reasons:
  - Gradle detected a problem with the following location: '/path/to/project/.editorconfig'. Reason: Task ':runKtlintFormatOverMainSourceSet' uses this output of task ':runKtlintFormatOverKotlinScripts' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0-milestone-3/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/path/to/project/build/intermediates/ktLint/reporters.bin'. Reason: Task ':runKtlintFormatOverMainSourceSet' uses this output of task ':runKtlintFormatOverKotlinScripts' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0-milestone-3/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/path/to/project/src/main/java'. Reason: Task ':runKtlintFormatOverMainSourceSet' uses this output of task ':runKtlintFormatOverKotlinScripts' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0-milestone-3/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/path/to/project/src/main/kotlin'. Reason: Task ':runKtlintFormatOverMainSourceSet' uses this output of task ':runKtlintFormatOverKotlinScripts' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0-milestone-3/userguide/validation_problems.html#implicit_dependency for more details about this problem.
Gradle detected a problem with the following location: '/path/to/project/.editorconfig'. Reason: Task ':runKtlintFormatOverMainSourceSet' uses this output of task ':runKtlintFormatOverKotlinScripts' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0-milestone-3/userguide/validation_problems.html#implicit_dependency for more details about this problem. This behaviour has been deprecated and is scheduled to be removed in Gradle 8.0. Execution optimizations are disabled to ensure correctness. See https://docs.gradle.org/7.0-milestone-3/userguide/more_about_tasks.html#sec:up_to_date_checks for more details.
Gradle detected a problem with the following location: '/path/to/project/build/intermediates/ktLint/reporters.bin'. Reason: Task ':runKtlintFormatOverMainSourceSet' uses this output of task ':runKtlintFormatOverKotlinScripts' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0-milestone-3/userguide/validation_problems.html#implicit_dependency for more details about this problem. This behaviour has been deprecated and is scheduled to be removed in Gradle 8.0. Execution optimizations are disabled to ensure correctness. See https://docs.gradle.org/7.0-milestone-3/userguide/more_about_tasks.html#sec:up_to_date_checks for more details.
Gradle detected a problem with the following location: '/path/to/project/src/main/java'. Reason: Task ':runKtlintFormatOverMainSourceSet' uses this output of task ':runKtlintFormatOverKotlinScripts' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0-milestone-3/userguide/validation_problems.html#implicit_dependency for more details about this problem. This behaviour has been deprecated and is scheduled to be removed in Gradle 8.0. Execution optimizations are disabled to ensure correctness. See https://docs.gradle.org/7.0-milestone-3/userguide/more_about_tasks.html#sec:up_to_date_checks for more details.
Gradle detected a problem with the following location: '/path/to/project/src/main/kotlin'. Reason: Task ':runKtlintFormatOverMainSourceSet' uses this output of task ':runKtlintFormatOverKotlinScripts' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0-milestone-3/userguide/validation_problems.html#implicit_dependency for more details about this problem. This behaviour has been deprecated and is scheduled to be removed in Gradle 8.0. Execution optimizations are disabled to ensure correctness. See https://docs.gradle.org/7.0-milestone-3/userguide/more_about_tasks.html#sec:up_to_date_checks for more details.

> Task :ktlintMainSourceSetFormat
Execution optimizations have been disabled for task ':ktlintMainSourceSetFormat' to ensure correctness due to the following reasons:
  - Gradle detected a problem with the following location: '/path/to/project/build/intermediates/ktLint/reporterProviders.bin'. Reason: Task ':ktlintMainSourceSetFormat' uses this output of task ':runKtlintFormatOverKotlinScripts' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0-milestone-3/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/path/to/project/build/intermediates/ktLint/reporters.bin'. Reason: Task ':ktlintMainSourceSetFormat' uses this output of task ':runKtlintFormatOverKotlinScripts' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0-milestone-3/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/path/to/project/build/intermediates/ktLint/runKtlintFormatOverMainSourceSet_errors.bin'. Reason: Task ':ktlintMainSourceSetFormat' uses this output of task ':runKtlintFormatOverKotlinScripts' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0-milestone-3/userguide/validation_problems.html#implicit_dependency for more details about this problem.
Gradle detected a problem with the following location: '/path/to/project/build/intermediates/ktLint/reporterProviders.bin'. Reason: Task ':ktlintMainSourceSetFormat' uses this output of task ':runKtlintFormatOverKotlinScripts' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0-milestone-3/userguide/validation_problems.html#implicit_dependency for more details about this problem. This behaviour has been deprecated and is scheduled to be removed in Gradle 8.0. Execution optimizations are disabled to ensure correctness. See https://docs.gradle.org/7.0-milestone-3/userguide/more_about_tasks.html#sec:up_to_date_checks for more details.
Gradle detected a problem with the following location: '/path/to/project/build/intermediates/ktLint/reporters.bin'. Reason: Task ':ktlintMainSourceSetFormat' uses this output of task ':runKtlintFormatOverKotlinScripts' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0-milestone-3/userguide/validation_problems.html#implicit_dependency for more details about this problem. This behaviour has been deprecated and is scheduled to be removed in Gradle 8.0. Execution optimizations are disabled to ensure correctness. See https://docs.gradle.org/7.0-milestone-3/userguide/more_about_tasks.html#sec:up_to_date_checks for more details.
Gradle detected a problem with the following location: '/path/to/project/build/intermediates/ktLint/runKtlintFormatOverMainSourceSet_errors.bin'. Reason: Task ':ktlintMainSourceSetFormat' uses this output of task ':runKtlintFormatOverKotlinScripts' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0-milestone-3/userguide/validation_problems.html#implicit_dependency for more details about this problem. This behaviour has been deprecated and is scheduled to be removed in Gradle 8.0. Execution optimizations are disabled to ensure correctness. See https://docs.gradle.org/7.0-milestone-3/userguide/more_about_tasks.html#sec:up_to_date_checks for more details.

> Task :runKtlintFormatOverTestSourceSet
Execution optimizations have been disabled for task ':runKtlintFormatOverTestSourceSet' to ensure correctness due to the following reasons:
  - Gradle detected a problem with the following location: '/path/to/project/.editorconfig'. Reason: Task ':runKtlintFormatOverTestSourceSet' uses this output of task ':runKtlintFormatOverKotlinScripts' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0-milestone-3/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/path/to/project/build/intermediates/ktLint/reporters.bin'. Reason: Task ':runKtlintFormatOverTestSourceSet' uses this output of task ':runKtlintFormatOverKotlinScripts' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0-milestone-3/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/path/to/project/src/test/java'. Reason: Task ':runKtlintFormatOverTestSourceSet' uses this output of task ':runKtlintFormatOverKotlinScripts' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0-milestone-3/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/path/to/project/src/test/kotlin'. Reason: Task ':runKtlintFormatOverTestSourceSet' uses this output of task ':runKtlintFormatOverKotlinScripts' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0-milestone-3/userguide/validation_problems.html#implicit_dependency for more details about this problem.
Gradle detected a problem with the following location: '/path/to/project/.editorconfig'. Reason: Task ':runKtlintFormatOverTestSourceSet' uses this output of task ':runKtlintFormatOverKotlinScripts' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0-milestone-3/userguide/validation_problems.html#implicit_dependency for more details about this problem. This behaviour has been deprecated and is scheduled to be removed in Gradle 8.0. Execution optimizations are disabled to ensure correctness. See https://docs.gradle.org/7.0-milestone-3/userguide/more_about_tasks.html#sec:up_to_date_checks for more details.
Gradle detected a problem with the following location: '/path/to/project/build/intermediates/ktLint/reporters.bin'. Reason: Task ':runKtlintFormatOverTestSourceSet' uses this output of task ':runKtlintFormatOverKotlinScripts' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0-milestone-3/userguide/validation_problems.html#implicit_dependency for more details about this problem. This behaviour has been deprecated and is scheduled to be removed in Gradle 8.0. Execution optimizations are disabled to ensure correctness. See https://docs.gradle.org/7.0-milestone-3/userguide/more_about_tasks.html#sec:up_to_date_checks for more details.
Gradle detected a problem with the following location: '/path/to/project/src/test/java'. Reason: Task ':runKtlintFormatOverTestSourceSet' uses this output of task ':runKtlintFormatOverKotlinScripts' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0-milestone-3/userguide/validation_problems.html#implicit_dependency for more details about this problem. This behaviour has been deprecated and is scheduled to be removed in Gradle 8.0. Execution optimizations are disabled to ensure correctness. See https://docs.gradle.org/7.0-milestone-3/userguide/more_about_tasks.html#sec:up_to_date_checks for more details.
Gradle detected a problem with the following location: '/path/to/project/src/test/kotlin'. Reason: Task ':runKtlintFormatOverTestSourceSet' uses this output of task ':runKtlintFormatOverKotlinScripts' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0-milestone-3/userguide/validation_problems.html#implicit_dependency for more details about this problem. This behaviour has been deprecated and is scheduled to be removed in Gradle 8.0. Execution optimizations are disabled to ensure correctness. See https://docs.gradle.org/7.0-milestone-3/userguide/more_about_tasks.html#sec:up_to_date_checks for more details.

> Task :ktlintTestSourceSetFormat
Execution optimizations have been disabled for task ':ktlintTestSourceSetFormat' to ensure correctness due to the following reasons:
  - Gradle detected a problem with the following location: '/path/to/project/build/intermediates/ktLint/reporterProviders.bin'. Reason: Task ':ktlintTestSourceSetFormat' uses this output of task ':runKtlintFormatOverKotlinScripts' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0-milestone-3/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/path/to/project/build/intermediates/ktLint/reporters.bin'. Reason: Task ':ktlintTestSourceSetFormat' uses this output of task ':runKtlintFormatOverKotlinScripts' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0-milestone-3/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/path/to/project/build/intermediates/ktLint/runKtlintFormatOverTestSourceSet_errors.bin'. Reason: Task ':ktlintTestSourceSetFormat' uses this output of task ':runKtlintFormatOverKotlinScripts' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0-milestone-3/userguide/validation_problems.html#implicit_dependency for more details about this problem.
Gradle detected a problem with the following location: '/path/to/project/build/intermediates/ktLint/reporterProviders.bin'. Reason: Task ':ktlintTestSourceSetFormat' uses this output of task ':runKtlintFormatOverKotlinScripts' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0-milestone-3/userguide/validation_problems.html#implicit_dependency for more details about this problem. This behaviour has been deprecated and is scheduled to be removed in Gradle 8.0. Execution optimizations are disabled to ensure correctness. See https://docs.gradle.org/7.0-milestone-3/userguide/more_about_tasks.html#sec:up_to_date_checks for more details.
Gradle detected a problem with the following location: '/path/to/project/build/intermediates/ktLint/reporters.bin'. Reason: Task ':ktlintTestSourceSetFormat' uses this output of task ':runKtlintFormatOverKotlinScripts' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0-milestone-3/userguide/validation_problems.html#implicit_dependency for more details about this problem. This behaviour has been deprecated and is scheduled to be removed in Gradle 8.0. Execution optimizations are disabled to ensure correctness. See https://docs.gradle.org/7.0-milestone-3/userguide/more_about_tasks.html#sec:up_to_date_checks for more details.
Gradle detected a problem with the following location: '/path/to/project/build/intermediates/ktLint/runKtlintFormatOverTestSourceSet_errors.bin'. Reason: Task ':ktlintTestSourceSetFormat' uses this output of task ':runKtlintFormatOverKotlinScripts' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0-milestone-3/userguide/validation_problems.html#implicit_dependency for more details about this problem. This behaviour has been deprecated and is scheduled to be removed in Gradle 8.0. Execution optimizations are disabled to ensure correctness. See https://docs.gradle.org/7.0-milestone-3/userguide/more_about_tasks.html#sec:up_to_date_checks for more details.

Execution optimizations have been disabled for 4 invalid unit(s) of work during this build to ensure correctness.
Please consult deprecation warnings for more details.

BUILD SUCCESSFUL in 5s
7 actionable tasks: 7 executed
```

### PR DESCRIPTION
**Pull request by KotlinIsland**
adds workaround for #457

### CODE REVIEW

**Review by Tapchicoma — COMMENTED** [review]
This is rather workaround then a good solution. Kotlin scripts and Kotlin sources format tasks (and generate reports) should not depend on each other and should be able to run in parallel.

**Comment by pvegh** [comment]
Can we merge this, save us from millions of warnings, and put lipstick on the pig later?

**Comment by Tapchicoma** [comment]
@pvegh yes, I will merge it and look into cause/better solution for `10.2.0` release

File: plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/TaskCreation.kt [hunk 1]
```diff
@@ -127,6 +135,7 @@ internal fun <T : BaseKtLintCheckTask> createGenerateReportsTask(
 ): TaskProvider<GenerateReportsTask> = pluginHolder.target.registerTask(
     GenerateReportsTask.generateNameForSourceSets(sourceSetName, lintType)
 ) {
+    dependsOn(project.tasks.named("runKtlintFormatOverKotlinScripts"))
```
Comment (davidburstromspotify) [inline plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/TaskCreation.kt:138]: You might not want to introduce an actual task dependency here, but rather enforce `mustRunAfter`. Please see the last sentences of https://docs.gradle.org/7.0.2/userguide/validation_problems.html#implicit_dependency

> In some cases, adding a dependency on the producing task is not desired, for example when the consumer generates reports for possible multiple tasks. In this case you can introduce an ordering between the two tasks by using Task.mustRunAfter().

**Comment by Tapchicoma** [comment]
FYI: @KotlinIsland seems your changelog check workflow is not working as expected. I would expect this PR will pass it as there is required entry in the changelog.

**Comment by KotlinIsland** [comment]
@Tapchicoma Sorry 😳here's a fix #493

**Pull request merged**